pkg/server/api/bundle/v1/service.go

```diff
@@ -518,3 +569,50 @@ func applyBundleMask(b *types.Bundle, mask *types.BundleMask) {
 		b.JwtAuthorities = nil
 	}
 }
+
+func fieldsFromBundleProto(proto *types.Bundle, inputMask *types.BundleMask) logrus.Fields {
+	fields := logrus.Fields{
+		telemetry.TrustDomainID: proto.TrustDomain,
+	}
+
+	if inputMask == nil || inputMask.RefreshHint {
+		fields[telemetry.RefreshHint] = proto.RefreshHint
+	}
+
+	if inputMask == nil || inputMask.SequenceNumber {
+		fields[telemetry.SequenceNumber] = proto.SequenceNumber
+	}
+
+	if inputMask == nil || inputMask.JwtAuthorities {
+		for k, v := range fieldsFromJwtAuthoritiesProto(proto.JwtAuthorities) {
+			fields[k] = v
+		}
+	}
+
+	if inputMask == nil || inputMask.X509Authorities {
+		for k, v := range fieldsFromX509AuthoritiesProto(proto.X509Authorities) {
+			fields[k] = v
+		}
+	}
+	return fields
+}
+
+func fieldsFromJwtAuthoritiesProto(jwtAuthorities []*types.JWTKey) logrus.Fields {
+	fields := make(logrus.Fields, 3*len(jwtAuthorities))
+	for i, jwtAuthority := range jwtAuthorities {
+		fields[fmt.Sprintf("jwt_authority_expires_at.%d", i)] = jwtAuthority.ExpiresAt
+		fields[fmt.Sprintf("jwt_authority_key_id.%d", i)] = jwtAuthority.KeyId
+		fields[fmt.Sprintf("jwt_authority_public_key.%d", i)] = api.HashByte(jwtAuthority.PublicKey)
```

Comment:
nit: use constants from telemetry package rather than redefining them here

Reply:
Also for the public key, can we call the log field key `jwt_authority_public_key_sha256` so it's more obvious that it's a hash?